### PR TITLE
UI: Stop job button

### PR DESCRIPTION
**Pull request by DingoEatingFuzz**
Introduces a stop button on all job detail pages.

stop-job-pr

The stop button is a new two-step confirmation button.

### CODE REVIEW

File: ui/app/templates/components/two-step-button.hbs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{{#if isIdle}}
+  <button class="button is-warning is-small is-inline" onclick={{action "promptForConfirmation"}}>
```
Comment (meirish) [inline ui/app/templates/components/two-step-button.hbs]: I think this needs type="button"

File: ui/app/templates/components/two-step-button.hbs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    {{idleText}}
+  </button>
+{{else if isPendingConfirmation}}
+  <span class="confirmation-text">{{confirmationMessage}}</span>
+  <button class="button is-dark is-outlined is-small is-inline" onclick={{action (queue
```
Comment (meirish) [inline ui/app/templates/components/two-step-button.hbs]: type here too please!

File: ui/app/templates/components/two-step-button.hbs (hunk 1 above)
Comment (DingoEatingFuzz) [reply ui/app/templates/components/two-step-button.hbs]: Thanks! Updated it.

**Comment by danlsgiga** [comment]
Thanks for the quick feedback @DingoEatingFuzz.
From my perspective, it does not make much sense to have an stop button in the UI if there are no plans to have a start button. 

I understand the fears around deviating from infrastructure as code and I truly endorse this vision.
The stop button does not change code, just the state of the job and that's fine!! But having the same behaviour on a start button (changing state, not code) would not hurt the infrastructure as code philosophy since we'll be basically resubmitting the same job that is already in nomad but just instructing it to run. 

Anyhow, just my .2 on the subject! Keep up the amazing work!!

**Pull request merged**